### ISSUE TITLE
Optional chaining is defined as TypeScript specific, but should it be?

### ISSUE THREAD

**Issue by dsherret** [swc-project/swc#525]
Right now, optional chaining is defined as a `TsOptionalChainingExpression`:

```rust
#[tag("TsOptionalChainingExpression")]
TsOptChain(TsOptChain),
```

But this is not TypeScript specific. Optional chaining is currently [stage 4 in ecmascript](https://github.com/tc39/proposal-optional-chaining).

Perhaps this should just be renamed to `OptionalChainingExpression`?

Related issue: #442

**Comment by kdy1** [comment]
@dsherret I'll change it soon. Btw, how can I know feature list of stage 4 / stage 3?
I want to implement all of them..

**Comment by kdy1** [comment]
@dsherret  Fixed with `swc_ecma_ast@0.13`, `swc_ecma_parser@0.15` and `swc_ecma_codegen@0.11`

**Comment by dsherret** [comment]
The proposals can be seen here: https://github.com/tc39/proposals

Specifically:

* https://github.com/tc39/proposals/blob/master/finished-proposals.md
* https://github.com/tc39/proposals#stage-3

I'll open up things as I come across them. Thanks again for all your hard work!
